### PR TITLE
py-pybind11: add v2.13.0 -> v2.13.4

### PR DESCRIPTION
**Pull request by stephenswat**
This PR adds py-pybind11 versions 2.13.0, 2.13.1, 2.13.2, 2.13.3, and 2.13.4. It also adds a new conflict between gcc 14 and pybind versions up to and including 2.13.1.

### CODE REVIEW

**Review by wdconinc — APPROVED** [review]
Verified checksums. Deferring to maintainer for final say-so.

**Comment by adamjstewart** [comment]
No response from maintainer, but I also need these new versions, so let's merge.

**Pull request merged**
